### PR TITLE
Few undefined behaviour issue in code

### PR DESCRIPTION
**Pull request by Kokan**
Compiling with `-fsanitize=undefined` and executing the UT test set reports these issues (well others as well, but most of them are related to nvtables)

- [ ] check zoneinfo specification for size limits (connected to unsigned/signed question)

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/timeutils/zoneinfo.c [hunk 1]
```diff
@@ -116,7 +116,7 @@ readcoded32(unsigned char **input, gint64 minv, gint64 maxv)
   *input += 4;
 
   for (i = 0, shift = 24; i < 4; ++i, shift -= 8)
-    val |= buf[i] << shift;
+    val |= (gint64)(buf[i]) << shift;
```
Comment (MrAnno) [inline lib/timeutils/zoneinfo.c]: Shifting a signed value can also be undefined if you reach the sign bit. Can this shift be done on an unsigned type instead?

Comment (Kokan) [reply lib/timeutils/zoneinfo.c]: In this case it cannot (reading 32 bit). But using unsigned might not hurt, I'll check it.

Comment (MrAnno) [reply lib/timeutils/zoneinfo.c]: Let's not block on this.  `shift` is less then 64bit; `buf[i]` is originally unsigned ( the value on the left-hand side is guaranteed to be positive), so this should be correct.

**Pull request merged**